corporate/lib/support.py

```diff
@@ -260,26 +261,46 @@ def get_plan_data_for_support_view(
         current_plan=plan,
     )
 
-    if plan is not None:
-        new_plan, last_ledger_entry = billing_session.make_end_of_cycle_updates_if_needed(
-            plan, timezone_now()
+    if plan_data.current_plan is not None:
+        last_ledger_entry = (
+            LicenseLedger.objects.filter(
+                plan=plan_data.current_plan, event_time__lte=timezone_now()
+            )
+            .order_by("-id")
+            .first()
         )
-        if last_ledger_entry is not None:
-            if new_plan is not None:
-                plan_data.current_plan = new_plan  # nocoverage
+
+        if last_ledger_entry is None:  # nocoverage
```

Comment:
It's very unlikely that this would ever occur as throughout the billing code/logic there are asserts for an existing LicenseLedger object for active plans, so we could alternatively just do that assertion here. The comment below explains a bit of my logic for not doing that.